#168367194 view a specific mentor endpoint

Mbonigabay opened the pull request:
#### What does this PR do?

- Build a view a specific mentor endpoint

#### Description of Task to be completed?

- Add a view specific mentor endpoint

#### How should this be manually tested?

- Use Postman to  test the endpoint

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [https://www.pivotaltracker.com/story/show/168367194](https://www.pivotaltracker.com/story/show/168367194)

#### Screenshots (if appropriate)
![ViewAMentor](https://user-images.githubusercontent.com/47386346/64769203-274afc80-d54b-11e9-974d-b5648d29a009.PNG)

#### Questions:

- N/A

Mbonigabay committed:
feature (view a specific mentor endpoint): view a specific mentor endpoint

-add a view specific mentor endpoint
-made one test

[Finishes #168367194]

Kasulejoseph commented on app/controller/userController.js:
```diff
@@ -122,5 +122,30 @@ class userController {
           }
   }
 
+  static async ViewAMentor(req, res) {
```
This should be in the mentor controller file, create a mentor controller file and move every controller related to mentors in that file

Kasulejoseph commented on app/controller/userController.js:
```diff
@@ -122,5 +122,30 @@ class userController {
+          try {
+            const mentor = await pool.query(users.searchMentorById, [req.params.id])
+            if(mentor.rowCount !== 0){
+            if (!mentor.error) {
```
Avoid these nested if, find better ways to implement this logic, Willy or Sadrack can help you with this.

Kasulejoseph commented on app/controller/userController.js:
```diff
@@ -122,5 +122,30 @@ class userController {
+            if (!mentor.error) {
+              return res.status(200).json({
+                status: 200,
+                data: mentor.rows,
```
You're missing `message` in the success response

Kasulejoseph commented on app/controller/userController.js:
```diff
@@ -122,5 +122,30 @@ class userController {
+            }}else {
+              return res.status(400).json({
+                status: 400,
+                message: 'no mentor by that id',
```
400 is not the right status code here..., also message should only be used for success responses, failed requests should have an `error` and not `message`

Kasulejoseph commented on app/controller/userController.js:
```diff
@@ -122,5 +122,30 @@ class userController {
+            }
+            return res.status(401).json({
+              status: 500,
+              message: 'server error please try again later',
```
Same here... should be an error, and the status code...

Mbonigabay committed:
chore (respond to feedback): making feedback

-create a mentor controller
-avoid nested if
-fix messages

[Finishes #168451467]

Kasulejoseph reviewed (approved):
Looks good to merge

Mbonigabay committed:
ch-fix-user-controller
